app/assets/javascripts/discourse/app/controllers/discovery/topics.js

```diff
@@ -127,12 +127,7 @@ const controllerOpts = {
       Topic.resetNew(this.category, !this.noSubcategories, {
```

Comment:
The tag page is another place where users can dismiss new topics and this function is used:

https://github.com/discourse/discourse/blob/20982ef875a149132be3ea04265a81377a5c6d76/app/assets/javascripts/discourse/app/controllers/tag-show.js#L95-L109

We need to update it so it works the same way as the global `/new` view.

Reply:
This should be fine, login to display modal or is hidden in `Topic.resetNew`

<img width="886" alt="Screenshot 2023-06-02 at 3 15 39 pm" src="https://github.com/discourse/discourse/assets/72780/4270bd29-3859-40bd-aceb-f3498da71260">

Reply:
Yes, the refactoring comment and also the `tag-show.js` controller still expects `Topic.resetNew` to return a promise, but with the changes in this PR `Topic.resetNew` doesn't always return a promise anymore.

![image](https://github.com/discourse/discourse/assets/17474474/3e4d4469-a5de-44cb-a009-67d10b8d1656)